src/api.ts

```diff
@@ -0,0 +1,374 @@
+    const path = `/sections/${sectionId}`,
+      params = { qs: { v: version, page, limit } };
+    const res = await this.callAPI("GET", path, params);
+    return res.section;
+  }
+
+  /**
+   * Utility function to fetch all items in a section, automatically paging if needed.
+   *
+   * The API limits fetches to 200. This function recursively calls fetchSection until all items have been retrieved.
+   * To page manually, use `fetchSection`
+   *
+   * @param sectionId the section uuid
+   * @param version the version number of the section to fetch
+   * @param limit The max number of items to fetch
+   * @returns A promise resolving the section and the items matching the page/limit
+   */
+  async fetchAllItemsInSection(sectionId: string, version = 0): Promise<Item[]> {
+    const limit = 200; // API Enforces <= 200
+    let page = 1,
+      results = [];
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const section = await this.fetchSection(sectionId, version, page, limit);
+      const items = section.items;
+      results = [...results, ...items];
+      if (items.length < limit) {
+        return results;
+      }
+      page += 1;
+    }
+  }
+
+  async fetchAssetsForHeading(sectionId: string, headingId: string, version = 0): Promise<Item[]> {
+    console.error("fetchAssetsForHeading() is deprecated, please use fetchItemsForHeading()");
+    return await this.fetchItemsForHeading(sectionId, headingId, version);
+  }
+
+  /**
+   * Fetch all items that fall under a given heading in a section
+   * @param sectionId the section id the heading is in
+   * @param headingId the uuid or string name of the the heading
+   * @param version the version number of the section to fetch
+   * @returns A promise resolving an array of items that fall under the desired heading. Can be empty.
+   *
+   * Note: If using the heading string, the first heading with that string will be used. sectionID is recommended to target a specific heading.
+   */
+  async fetchItemsForHeading(sectionId: string, headingId: string, version = 0): Promise<Item[]> {
+    let page = 1;
+    let found = false;
+    const results = [];
+
+    function isMatch(item: Item) {
+      return (
+        item.type === ItemType.Heading && (item.data.content === headingId || item.id === headingId)
+      );
+    }
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const section = await this.fetchSection(sectionId, version, page, 200);
+
+      const items = section.items;
+      const count = items.length;
+      if (count == 0) {
+        return results;
+      }
+      for (let idx = 0; idx < count; idx++) {
+        const item = items[idx];
+        if (item.type == ItemType.Heading && found) {
+          return results;
+        } else if (isMatch(item)) {
+          found = true;
+        } else if (found) {
+          results.push(item);
+        }
+      }
+      page += 1;
```

Comment:
What's the purpose of the `while(true)` ?

Reply:
If I'm understanding this correctly, wouldn't we want something like `while (page <= totalPages)`

Reply:
This is essentially saying, keep running forever until we break out. So the if `count == 0` is effectively doing the same job as what you are suggesting but without having to determine the total number of pages and set a variable outside the loop which in itself could be buggy if the count is wrong or you set that variable incorrectly.